README.md

```diff
@@ -18,25 +20,185 @@ $ npm install --save pipage
+
+// Find a stream in the pipeline:
+var index = pipeline.indexOf( stream )
+var lastIndex = pipeline.lastIndexOf( stream )
+```
+
+### Example
+
+```js
+var Pipage = require('pipage')
+var path = require('path')
+var zlib = require('zlib')
+var unbzip2 = require('unbzip2-stream')
+var xz = require('xz')
+
+var pipeline = new Pipage()
+
+switch( path.extname( filename ) ) {
+  case '.gz':
+    pipeline.prepend( zlib.createUnzip() );
+    break
+  case '.bz':
+  case '.bz2':
+    pipeline.prepend( unbzip2() );
+    break
+  case '.xz':
+    pipeline.prepend( new xz.Decompressor() )
+    break
 }
 
-new BlockReadStream( '/path/to/resin-os.img' )
+fs.createReadStream( filename )
   .pipe( pipeline )
-  .pipe( new BlockWriteStream( '/dev/rdisk2' ) )
-  .once( 'finish', function() {
-    // happy face
-  })
+  .pipe( fs.createWriteStream( destination ) )
+```
+
+### Events
+
+#### Error handling
+
+Errors from streams within the pipeline are listened to,
+and re-emitted on the pipeline itself, with an additional `.stream` property
+being set on the `error` object, which is the stream that emitted it:
+
+```js
+pipeline.on( 'error', function( error ) {
+  // This error originated from the pipeline's internal stream
+  // available as `error.stream`, not from the pipeline itself
+  if( error.stream ) {
+    // ...
+  } else {
+    // This error came from the pipeline itself
+  }
+})
+```
+
+#### Binding to contained stream's events
+
+```js
+var stream = pipeline.get(-1)
+
+// Re-emit a stream's events on the pipeline:
+pipeline.bind( stream, 'eventname' )
+pipeline.bind( stream, [ 'someevent', 'otherevent' ])
+
+// Remove re-emission of a stream's event:
+pipeline.unbind( stream, 'eventname' )
+
+// Stop re-emission of all of stream's events on the pipeline:
+pipeline.unbindAll( stream )
+```
+
+##### Example
+
+```js
+var Pipage = require('pipage')
+
+// Let's say we have a stream which emits an event
+// we want to capture without having to get a reference to
+// that particular stream:
+module.exports = function createPipeline() {
+
+  var checksumStream = createChecksumStream( 'sha256', 'md5' )
+  var pipeline = new Pipage([ checksumStream ])
+
+  // This will cause the pipeline to re-emit
+  // the 'checksums' event from the `checksumStream`
+  pipeline.bind( checksumStream, 'checksums' )
+
+  // Add some more fancy things to the pipeline...
+
+  return pipeline
+
+}
+```
+
+```js
+var createPipeline = require('./create-pipeline')
+var pipeline = createPipeline()
+
+// Now we can listen for the bound event directly on the pipeline
+pipeline.on( 'checksums', function( checksums ) {
+  // Validate the checksums, etc...
+})
+
+fs.createReadStream( filename )
+  .pipe( pipeline )
+  .resume()
+```
+
+### Nested Pipelines
+
+Since pipelines are duplex streams, and contain duplex streams, they can be nested arbitrarily:
+
+```js
+var pipeline = new Pipage([
+  new Pipage(),
+  new Pipage([
+    new Pipage()
+  ])
+])
```

Comment:
A pipeline == a [duplex stream](https://nodejs.org/api/stream.html#stream_types_of_streams), which means it behaves just like any other readable & writable stream (and if data wasn't flowing in a linear fashion, then it wouldn't be a stream, right?). The nesting here is no different than having any other stream in the pipeline, and is just meant to demonstrate that this works as expected.

> P.S. Should the length property be added to the API docs? Should there be some public method of enumerating the custom events bound to a pipeline?

No, I don't think so – the code utilizing this needs to know which events it binds & uses, just like with any other EventEmitter.

> Hmm, I wonder if something like this should/could be better supported by a method that did something like:

Pipelines are meant to be like modules – reasonably opaque to their consumer. The consuming streams will usually not faff about with the pipeline's internals. However, I've already noted intent to implement nested access [in a few "todo" items](https://github.com/resin-io-modules/pipage/tree/master/doc#pipelinegetindex--stream), but that's not the topic of this PR.

> I've got no idea what kind of mangling would be needed to transfer all the custom-bound events over though... (or even if this crazy suggestion actually makes sense?!?)

There no need to transfer any events, as the pipelines behave like regular streams. If there's anything other than the streamed data you'd want to get out of it, you'd have to add an event handler just the same as with everything else.